tools/monorepo-utils/src/md-docs/lib/generate-manifest.ts

```diff
@@ -23,59 +28,28 @@ function generatePageId( filePath: string, prefix = '' ) {
 	return hash.digest( 'hex' );
 }
 
-/**
- *	Generates a file url relative to the root directory provided.
- *
- * @param baseUrl          The base url to use for the file url.
- * @param rootDirectory    The root directory where the file resides.
- * @param subDirectory     The sub-directory where the file resides.
- * @param absoluteFilePath The absolute path to the file.
- * @return The file url.
- */
-export const generateFileUrl = (
-	baseUrl: string,
-	rootDirectory: string,
-	subDirectory: string,
-	absoluteFilePath: string
-) => {
-	// check paths are absolute
-	for ( const filePath of [
-		rootDirectory,
-		subDirectory,
-		absoluteFilePath,
-	] ) {
-		if ( ! path.isAbsolute( filePath ) ) {
-			throw new Error(
-				`File URLs cannot be generated without absolute paths. ${ filePath } is not absolute.`
-			);
-		}
-	}
-	// Generate a path from the subdirectory to the file path.
-	const relativeFilePath = path.resolve( subDirectory, absoluteFilePath );
-
-	// Determine the relative path from the rootDirectory to the filePath.
-	const relativePath = path.relative( rootDirectory, relativeFilePath );
-
-	return `${ baseUrl }/${ relativePath }`;
-};
-
 async function processDirectory(
 	rootDirectory: string,
 	subDirectory: string,
 	projectName: string,
 	baseUrl: string,
+	baseEditUrl: string,
 	checkReadme = true
 ): Promise< Category > {
-	let category: Category = {};
+	const category: Category = {};
 
 	// Process README.md (if exists) for the category definition.
 	const readmePath = path.join( subDirectory, 'README.md' );
 
 	if ( checkReadme && fs.existsSync( readmePath ) ) {
 		const readmeContent = fs.readFileSync( readmePath, 'utf-8' );
-		const readmeFrontmatter = matter( readmeContent ).data;
-		category = { ...readmeFrontmatter };
-		category.posts = [];
+		const frontMatter = generatePostFrontMatter( readmeContent );
+		Object.assign( category, frontMatter );
+	} else if ( checkReadme ) {
+		// derive the category title from the directory name, capitalize first letter
```

Comment:
Since we already care about capitalization here we might consider going with something approximating some version of [title case](https://en.wikipedia.org/wiki/Title_case).

Reply:
To do that we need to do a best effort to split up the directory name by dashes or underscores or some other convention. How about i address some improvements to this in a follow up?

Reply:
Resolved here https://github.com/woocommerce/woocommerce/pull/39214/commits/85ac3e04db0e368443f24bb5d6a2ebe4493ae7cc

Reply:
Sounds good to me, thanks!